flake-modules/dev/launch-test.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+: "${NIXVIM_NIX_COMMAND:=nix}"
+if [[ -z $NIXVIM_SYSTEM ]]; then
+	NIXVIM_SYSTEM=$(nix eval --raw --impure --expr "builtins.currentSystem")
+fi
+
+help() {
+	cat <<EOF
+Usage: tests [OPTIONS] [tests...]
+
+If tests are passed on the command line only these will be launched
+
+Options:
+	-h/--help: Display this help message and exit
+	-l/--list: Display the list of tests and exit
+	-i/--interactive: Pick interactively the tests. Can't be supplied if tests where passed.
+EOF
+}
+
+if ! OPTS=$(getopt -o "hli" -l "help,list,interactive" -- "$@"); then
+	echo "Invalid options" >&2
+	help
+	exit 1
+fi
+
+eval set -- "$OPTS"
+
+specified_tests=()
+interactive=false
+
+mk_test_list() {
+	nix eval ".#checks.${NIXVIM_SYSTEM}" --apply builtins.attrNames --json |
+		jq -r 'map(select(startswith("test-")))[]'
+}
+
+while true; do
+	case "$1" in
+	-h | --help)
+		help
+		exit 0
+		;;
+	-l | --list)
+		mk_test_list
+		exit 0
+		;;
+	-i | --interactive)
+		interactive=true
+		shift
+		;;
+	--)
+		shift
+		specified_tests=("$@")
+		break
+		;;
+	esac
+done
+
+run_tests() {
+	# Add the prefix "checks.${system}." to each argument
+	if ! "${NIXVIM_NIX_COMMAND}" build -f . "${@/#/checks.${NIXVIM_SYSTEM}.}"; then
```

Comment:
Can we use `--file` instead of `-f`? I prefer using long-form options in scripts since anyone can follow their meaning, even if unfamiliar with the specific command's options.

Reply:
Also, does `nix build -f .` produce `result`? I'm wondering if we can speed up the generate-files build by doing all builds in one step?

Reply:
It does for each of the derivations passed to the command line.
I don't see how can be do all the builds in "one" step, I already pass all the builds to a single `nix` invocation :/

Reply:
Sorry, which comment are you replying to?

Reply:
> I'm wondering if we can speed up the generate-files build by doing all builds in one step?